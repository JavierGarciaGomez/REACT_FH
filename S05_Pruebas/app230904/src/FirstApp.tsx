type Props = { title: string; subtitle: string; name: string };

export const FirstApp = ({ title, subtitle, name }: Props) => (
  <>
    <h1 data-testid="test-title"> {title} </h1>
    {/* <code>{ JSON.stringify( newMessage ) }</code> */}
    <p>{subtitle}</p>
    <p>{subtitle}</p>
    <p>{name}</p>
  </>
);
