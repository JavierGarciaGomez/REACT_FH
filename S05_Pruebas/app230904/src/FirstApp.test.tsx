import { render } from "@testing-library/react";
import { FirstApp } from "./FirstApp";

describe("pruebas en <FirstApp />", () => {
  it("should match the snapshot", () => {
    const { container } = render(
      <FirstApp title={"a Title"} subtitle={"a subtitle"} name={"a name"} />
    );
    expect(container).toMatchSnapshot();
  });
  it("should have an h1 title", () => {
    const title = "a Title";

    const { getByText, container } = render(
      <FirstApp title={"a Title"} subtitle={"a subtitle"} name={"a name"} />
    );
    const h1 = container.querySelector("h1");

    expect(getByText(title)).toBeTruthy();
    expect(h1?.innerHTML).toContain(title);
  });

  it("should have an h1 title. Selecting element by id", () => {
    const title = "a Title";

    const { getByTestId } = render(
      <FirstApp title={"a Title"} subtitle={"a subtitle"} name={"a name"} />
    );
    expect(getByTestId("test-title").innerHTML).toContain(title);
  });
});
