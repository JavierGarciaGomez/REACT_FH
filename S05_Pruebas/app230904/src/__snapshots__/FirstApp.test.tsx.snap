// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <FirstApp /> should match the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    a Title
     
  </h1>
  <p>
    a subtitle
  </p>
  <p>
    a subtitle
  </p>
  <p>
    a name
  </p>
</div>
`;
